SpringProject/src/main/java/com/gradproject2019/users/service/UserServiceImpl.java

```diff
@@ -117,8 +114,28 @@ private void checkIfManagerExists(Long managerId) {
     }
 
     private void checkTokenMatchesUser(UUID token, Long userId) {
-        if(!authServiceImpl.getTokenById(token).getUserId().equals(userId)) {
-            throw new UserUnauthorisedException();
+        if(!tokenMatchesUser(token, userId)) {
+            throw new UserRecognisedButUnauthorizedException();
+        }
+    }
+
+    private boolean tokenMatchesUser(UUID token, Long userId) {
+        return authServiceImpl.getTokenById(token).getUserId().equals(userId);
+    }
+
+    private boolean userIdMatchesManagerId(User requestingUser, User requestedUser) {
+        return requestedUser.getManagerId() != null && requestingUser.getId().equals(requestedUser.getManagerId());
+    }
+
+    private boolean managerIdMatchesUserId(User requestingUser, User requestedUser) {
+        return requestingUser.getManagerId() != null && requestingUser.getManagerId().equals(requestedUser.getId());
+    }
+
+    private void checkUserRequestingIsAuthorized(Long requestedUserId, UUID token) {
+        User requestingUser = getUserById(authServiceImpl.getTokenById(token).getUserId());
+        User requestedUser = getUserById(requestedUserId);
+        if (!tokenMatchesUser(token, requestedUserId) && !userIdMatchesManagerId(requestingUser, requestedUser) && !managerIdMatchesUserId(requestingUser, requestedUser)) {
```

Comment:
You can call checkTokenMatches user here instead of tokenMachesUser and then you won't need tokenMatchesUser as a seperate method?

Is this the only place where userIdMatchesManagerId and managerIdMatchesUserId are called? If it is, then you can put the code here and delete those methods because their names are a bit confusing.